client/request.go

```diff
@@ -134,6 +134,9 @@ func (cli *Client) sendClientRequest(ctx context.Context, method, path string, q
 				return serverResp, ErrorConnectionFailed(cli.host)
 			}
 			if !err.Temporary() {
+				if strings.Contains(err.Error(), "connect: permission denied") {
```

Comment:
@cpuguy83 I added this change and the code looks a lot better. I also changed the PermissionDenied method to just a var with a errors.new call. To match what @thaJeztah was talking about. What do you guys think?